### PR TITLE
MAGN-9428: API Documentation - Priority 1

### PR DESCRIPTION
**Pull request by aosyatnik**
### Purpose

Link to YouTrack:
[MAGN-9428](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-9428) API Documentation - Priority 1

Link to doc: https://docs.google.com/spreadsheets/d/1PT8Ha_7764ZP9CH6JKG0Y2do38xT-pDjf1XyEbsqxKs/edit#gid=0

### Declarations

- [x] The code base is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.

### Reviewers

@ramramps 

### FYIs

@kronz 
@YanmengLi

### CODE REVIEW

**Comment by ramramps** [comment]
@aosyatnik - I will take this one.

File: src/DynamoCore/Graph/ModelBase.cs [hunk 1]
```diff
@@ -40,6 +43,10 @@ public UpdateValueParams(string propertyName, string propertyValue, ElementResol
         public ElementResolver ElementResolver { get; private set; }
     }
 
+    /// <summary>
+    /// Abstract class, that is used as base class for all objects with which user can interact in Dynamo.
```
Comment (ramramps) [inline src/DynamoCore/Graph/ModelBase.cs]: Say - "Base class for all objects with which user can interact in Dynamo".

File: src/DynamoCore/Graph/ModelBase.cs [hunk 2]
```diff
@@ -40,6 +43,10 @@ public UpdateValueParams(string propertyName, string propertyValue, ElementResol
...
+    /// Objects like nodes, connectors, ports, annotations etc.
```
Comment (ramramps) [inline src/DynamoCore/Graph/ModelBase.cs]: Remove this.

File: src/DynamoCore/Graph/ModelBase.cs [hunk 3]
```diff
@@ -121,31 +135,29 @@ public virtual double Height
...
-        public event EventHandler Updated; 
-        public void OnUpdated(EventArgs e)
-        {
-            if (Updated != null)
-                Updated(this, e);
+            get { return new Rect2D(x, y, width, height); }
         }
 
+        /// <summary>
+        /// Bool value, that indicates is object is selected or not.
```
Comment (ramramps) [inline src/DynamoCore/Graph/ModelBase.cs]: Say -  Returns true if the object is selected.

File: src/DynamoCore/Graph/ModelBase.cs [hunk 4]
```diff
@@ -173,21 +188,34 @@ public Guid GUID
...
+        /// <summary>
+        /// Protected constructor.
+        /// </summary>
         protected ModelBase()
         {
             GUID = Guid.NewGuid();
         }
 
+        /// <summary>
+        /// Selects object.
```
Comment (ramramps) [inline src/DynamoCore/Graph/ModelBase.cs]: Say - "this method sets the isSelected property on the object".
provide an example here. Like node.Select().

File: src/DynamoCore/Graph/ModelBase.cs [hunk 5]
```diff
@@ -173,21 +188,34 @@ public Guid GUID
...
+        /// </summary>
         public virtual void Select()
         {
             IsSelected = true;
         }
 
+        /// <summary>
+        /// Deselects object.
```
Comment (ramramps) [inline src/DynamoCore/Graph/ModelBase.cs]: Say - "this method sets the isSelected property on the object".

File: src/DynamoCore/Graph/ModelBase.cs [hunk 6]
```diff
@@ -206,15 +234,27 @@ public void SetSize(double w, double h)
             RaisePropertyChanged("Position");
         }
 
+        /// <summary>
+        /// Notifies listeners that is going to be disposed,
```
Comment (ramramps) [inline src/DynamoCore/Graph/ModelBase.cs]: Say - Invokes Dispose on the object.

File: src/DynamoCore/Graph/ModelBase.cs [hunk 7]
```diff
@@ -206,15 +234,27 @@ public void SetSize(double w, double h)
             RaisePropertyChanged("Position");
         }
 
+        /// <summary>
+        /// Notifies listeners that is going to be disposed, 
+        /// so that they can free, release, or reset their resources.
```
Comment (ramramps) [inline src/DynamoCore/Graph/ModelBase.cs]: remove this.

File: src/DynamoCore/Graph/ModelBase.cs [hunk 8]
```diff
@@ -206,15 +234,27 @@ public void SetSize(double w, double h)
...
             if (handler != null)
+            {
                 handler(this);
+            }
         }
 
         #region Command Framework Supporting Methods
 
+        /// <summary>
+        /// Calls protected UpdateValueCore, which updates some property of object.
```
Comment (ramramps) [inline src/DynamoCore/Graph/ModelBase.cs]: Say - Updates object properties.

File: src/DynamoCore/Graph/ModelBase.cs [hunk 9]
```diff
@@ -278,13 +329,17 @@ protected virtual XmlElement CreateElement(XmlDocument xmlDocument, SaveContext
...
+
         protected abstract void SerializeCore(XmlElement element, SaveContext context);
         protected abstract void DeserializeCore(XmlElement nodeElement, SaveContext context);
 
         #endregion
 
         #region ILogSource implementation
+
+        /// <summary>
+        /// Event, which is fired, when message should be logged.
```
Comment (ramramps) [inline src/DynamoCore/Graph/ModelBase.cs]: This event is fired when a message is logged.

File: src/DynamoCore/Graph/Nodes/Attributes.cs [hunk 10]
```diff
@@ -18,20 +18,40 @@ public NodeNameAttribute(string elementName)
         public string Name { get; set; }
     }
 
+    /// <summary>
+    ///     The NodeCategoryAttribute attribute allows the node implementer
```
Comment (ramramps) [inline src/DynamoCore/Graph/Nodes/Attributes.cs]: The NodeCategoryAttribute attribute allows the node implementer to define in which category the node will appear

File: src/DynamoCore/Graph/Nodes/Attributes.cs [hunk 11]
```diff
@@ -18,20 +18,40 @@ public NodeNameAttribute(string elementName)
...
         }
 
+        /// <summary>
+        /// Full name of the category. E.g. Core.List.Create
+        /// </summary>
         public string ElementCategory { get; set; }
     }
 
+    /// <summary>
+    ///     The NodeCategoryAttribute attribute allows the node implementer
```
Comment (ramramps) [inline src/DynamoCore/Graph/Nodes/Attributes.cs]: The NodeSearchTagsAttribute attribute allows the node implementer to  define search tags

File: src/DynamoCore/Graph/Nodes/Attributes.cs [hunk 12]
```diff
@@ -77,14 +105,27 @@ public IsInteractiveAttribute(bool isInteractive)
         public bool IsInteractive { get; set; }
```
Comment (ramramps) [inline src/DynamoCore/Graph/Nodes/Attributes.cs]: Comments?

File: src/DynamoCore/Graph/Nodes/Attributes.cs [hunk 13]
```diff
@@ -77,14 +105,27 @@ public IsInteractiveAttribute(bool isInteractive)
...
     }
 
+    /// <summary>
+    ///     The NodeCategoryAttribute attribute allows the node implementer
```
Comment (ramramps) [inline src/DynamoCore/Graph/Nodes/Attributes.cs]: NodeCategoryAttribute  = NodeDescriptionAttribute

File: src/DynamoCore/Graph/Nodes/Attributes.cs [hunk 14]
```diff
@@ -77,14 +105,27 @@ public IsInteractiveAttribute(bool isInteractive)
...
+    ///     to define node description.
+    /// </summary>
     [AttributeUsage(AttributeTargets.All)]
     public class NodeDescriptionAttribute : Attribute
     {
+        /// <summary>
+        /// Creates NodeDescriptionAttribute with string.
```
Comment (ramramps) [inline src/DynamoCore/Graph/Nodes/Attributes.cs]: Creates NodeDescriptionAttribute

File: src/DynamoCore/Graph/Nodes/Attributes.cs [hunk 15]
```diff
@@ -77,14 +105,27 @@ public IsInteractiveAttribute(bool isInteractive)
...
+        /// </summary>
+        /// <param name="description">String description</param>
```
Comment (ramramps) [inline src/DynamoCore/Graph/Nodes/Attributes.cs]: Description - Give an example.

File: src/DynamoCore/Graph/Nodes/Attributes.cs [hunk 16]
```diff
@@ -103,6 +144,9 @@ public NodeDescriptionAttribute(string descriptionResourceID, Type resourceType)
             }
         }
 
+        /// <summary>
+        /// Description of node.
```
Comment (ramramps) [inline src/DynamoCore/Graph/Nodes/Attributes.cs]: May be an example here?

File: src/DynamoCore/Graph/Nodes/Attributes.cs [hunk 17]
```diff
@@ -129,18 +173,26 @@ public NodeTypeIdAttribute(string description)
     }
 
     /// <summary>
-    ///     The DoNotLoadOnPlatforms attribute allows the node implementor
+    ///     The DoNotLoadOnPlatforms attribute allows the node implementer
     ///     to define an array of contexts in which the node will not
     ///     be loaded.
```
Comment (ramramps) [inline src/DynamoCore/Graph/Nodes/Attributes.cs]: remove in which the node will not  be loaded

File: src/DynamoCore/Graph/Nodes/NodeModel.cs [hunk 18]
```diff
@@ -2131,6 +2131,10 @@ public enum LacingStrategy
         CrossProduct
     };
 
+    /// <summary>
+    /// Enum, that is used when mouse enters, leaves or clicks port.
```
Comment (ramramps) [inline src/DynamoCore/Graph/Nodes/NodeModel.cs]: Defines Enum for Mouse events.
Used in port snapping.

File: src/DynamoCore/Graph/Nodes/PortModel.cs [hunk 19]
```diff
@@ -14,6 +14,10 @@ namespace Dynamo.Graph.Nodes
     /// </summary>
     public enum PortType { Input, Output };
 
+    /// <summary>
+    /// PortModel represents Dynamo ports. Dynamo port is little square on node.
```
Comment (ramramps) [inline src/DynamoCore/Graph/Nodes/PortModel.cs]: PortModel represents Dynamo ports.

File: src/DynamoCore/Graph/Nodes/PortModel.cs [hunk 20]
```diff
@@ -14,6 +14,10 @@ namespace Dynamo.Graph.Nodes
...
+    /// User can connect and disconnect ports and therefore change value of the node.
```
Comment (ramramps) [inline src/DynamoCore/Graph/Nodes/PortModel.cs]: Remove

File: src/DynamoCore/Graph/Nodes/PortModel.cs [hunk 21]
```diff
@@ -24,43 +28,70 @@ public class PortModel : ModelBase
 
         #region public members
 
+        /// <summary>
+        /// Wires,that are connected to port.
```
Comment (ramramps) [inline src/DynamoCore/Graph/Nodes/PortModel.cs]: Returns the connectors between the specified ports.

File: src/DynamoCore/Graph/Nodes/PortModel.cs [hunk 22]
```diff
@@ -24,43 +28,70 @@ public class PortModel : ModelBase
...
                 string useDefaultArgument = string.Empty;
                 if (!UsingDefaultValue && DefaultValueEnabled)
                     useDefaultArgument = " " + Properties.Resources.DefaultValueDisabled;
-                return portData.ToolTipString + useDefaultArgument; 
+                return portData.ToolTipString + useDefaultArgument;
             }
         }
 
+        /// <summary>
+        /// Type of the port.
```
Comment (ramramps) [inline src/DynamoCore/Graph/Nodes/PortModel.cs]: Example.

File: src/DynamoCore/Graph/Nodes/PortModel.cs [hunk 23]
```diff
@@ -24,43 +28,70 @@ public class PortModel : ModelBase
...
+        /// </summary>
         public PortType PortType
         {
-            get; private set;
+            get;
+            private set;
         }
 
+        /// <summary>
+        /// Node, that owns this port.
```
Comment (ramramps) [inline src/DynamoCore/Graph/Nodes/PortModel.cs]: Returns the Node.

File: src/DynamoCore/Graph/Nodes/PortModel.cs [hunk 24]
```diff
@@ -24,43 +28,70 @@ public class PortModel : ModelBase
...
+        /// <summary>
+        /// Index of the port.
+        /// </summary>
         public int Index
         {
             get { return Owner.GetPortModelIndex(this); }
         }
 
+        /// <summary>
+        /// LineIndex is used just in code block nodes.
```
Comment (ramramps) [inline src/DynamoCore/Graph/Nodes/PortModel.cs]: remove - just.

File: src/DynamoCore/Graph/Nodes/PortModel.cs [hunk 25]
```diff
@@ -24,43 +28,70 @@ public class PortModel : ModelBase
...
+        /// A code block node may 
+        /// contain empty lines in it, resulting in one PortModel being spaced out
```
Comment (ramramps) [inline src/DynamoCore/Graph/Nodes/PortModel.cs]: Remove all this.

File: src/DynamoCore/Graph/Nodes/PortModel.cs [hunk 26]
```diff
@@ -139,13 +170,25 @@ public AssociativeNode DefaultValue
...
+            get;
+            private set;
         }
 
-        public SnapExtensionEdges extensionEdges { get; set; }        
-    
+        /// <summary>
+        /// Based on extensionEdges port is aligned in UI.
+        /// Because of this alignment user can snap ports.
+        /// I.e. connect port even if he didn't click on port directly.
```
Comment (ramramps) [inline src/DynamoCore/Graph/Nodes/PortModel.cs]: Remove  /// I.e. connect port even if he didn't click on port directly

File: src/DynamoCore/Graph/Nodes/PortModel.cs [hunk 27]
```diff
@@ -139,13 +170,25 @@ public AssociativeNode DefaultValue
...
-            get; private set;
+            get;
+            private set;
         }
 
-        public SnapExtensionEdges extensionEdges { get; set; }        
-    
+        /// <summary>
+        /// Based on extensionEdges port is aligned in UI.
+        /// Because of this alignment user can snap ports.
```
Comment (ramramps) [inline src/DynamoCore/Graph/Nodes/PortModel.cs]: remove.

File: src/DynamoCore/Graph/Nodes/PortModel.cs [hunk 28]
```diff
@@ -254,17 +297,53 @@ protected override void DeserializeCore(XmlElement nodeElement, SaveContext cont
...
+        /// </summary>
         public string ToolTipString { get; set; }
+
+        /// <summary>
+        /// Default value of the port.
+        /// </summary>
         public AssociativeNode DefaultValue { get; set; }
+
+        /// <summary>
+        /// Property, that is used in code block nodes.
```
Comment (ramramps) [inline src/DynamoCore/Graph/Nodes/PortModel.cs]: This property is used in code block nodes.

File: src/DynamoCore/Graph/Nodes/PortModel.cs [hunk 29]
```diff
@@ -254,17 +297,53 @@ protected override void DeserializeCore(XmlElement nodeElement, SaveContext cont
...
+        /// A code block node may
```
Comment (ramramps) [inline src/DynamoCore/Graph/Nodes/PortModel.cs]: Remove all these lines.

File: src/DynamoCore/Graph/Notes/NoteModel.cs [hunk 30]
```diff
@@ -4,9 +4,18 @@
 
 namespace Dynamo.Graph.Notes
 {
+    /// <summary>
+    /// NoteModel represents notes in Dynamo.
+    /// Dynamo note is little square, where you can write something,
```
Comment (ramramps) [inline src/DynamoCore/Graph/Notes/NoteModel.cs]: Remove all the lines from DynamoNote. Just say NoteModel represents notes in Dynamo

File: src/DynamoCore/Graph/Notes/NoteModel.cs [hunk 31]
```diff
@@ -4,9 +4,18 @@
...
+    /// Dynamo note is little square, where you can write something, 
+    /// that you want to remember or pay someone's attention.
+    /// </summary>
     public class NoteModel : ModelBase
     {
-        private string text;      
+        private string text;
+      
+        /// <summary>
+        /// Text, that is inside note.
```
Comment (ramramps) [inline src/DynamoCore/Graph/Notes/NoteModel.cs]: Returns the text inside the note.

File: src/DynamoCore/Interfaces/IDynamoModel.cs [hunk 32]
```diff
@@ -10,17 +10,52 @@ namespace Dynamo.Interfaces
 {
     // TODO:http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-8116
 
+    /// <summary>
+    /// Interface contains definitions for core model of Dynamo.
+    /// </summary>
     public interface IDynamoModel : INotifyPropertyChanged
     {
+        /// <summary>
+        /// Opened workspaces in Dynamo.
```
Comment (ramramps) [inline src/DynamoCore/Interfaces/IDynamoModel.cs]: Represents the Workspaces in Dynamo.

File: src/DynamoCore/Interfaces/IDynamoModel.cs [hunk 33]
```diff
@@ -10,17 +10,52 @@ namespace Dynamo.Interfaces
...
+        /// </summary>
         IEnumerable<WorkspaceModel> Workspaces { get; }
 
+        /// <summary>
+        /// Workspace, that is currently opened.
```
Comment (ramramps) [inline src/DynamoCore/Interfaces/IDynamoModel.cs]: Represents the current workspace.

File: src/DynamoCore/Interfaces/IDynamoModel.cs [hunk 34]
```diff
@@ -10,17 +10,52 @@ namespace Dynamo.Interfaces
...
+        /// </summary>
         WorkspaceModel CurrentWorkspace { get; }
 
+        /// <summary>
+        /// Event, which is fired when workspace is cleared.
```
Comment (ramramps) [inline src/DynamoCore/Interfaces/IDynamoModel.cs]: This event is fired when workspace is cleared.

File: src/DynamoCore/Interfaces/IDynamoModel.cs [hunk 35]
```diff
@@ -10,17 +10,52 @@ namespace Dynamo.Interfaces
...
+        /// </summary>
         event Action<WorkspaceModel> WorkspaceCleared;
+
+        /// <summary>
+        /// Event, which is fired when Dynamo starting shuting down.
```
Comment (ramramps) [inline src/DynamoCore/Interfaces/IDynamoModel.cs]: This event is fired when Dynamo is shutting down.

File: src/DynamoCore/Interfaces/IDynamoModel.cs [hunk 36]
```diff
@@ -10,17 +10,52 @@ namespace Dynamo.Interfaces
...
+        /// </summary>
         event DynamoModelHandler ShutdownStarted;
+
+        /// <summary>
+        /// Event, which is fired when graph evaluation is comleted.
```
Comment (ramramps) [inline src/DynamoCore/Interfaces/IDynamoModel.cs]: This event is fired when graph evaluation is completed.

File: src/DynamoCore/Interfaces/IDynamoModel.cs [hunk 37]
```diff
@@ -10,17 +10,52 @@ namespace Dynamo.Interfaces
...
+        /// </summary>
         event EventHandler<EvaluationCompletedEventArgs> EvaluationCompleted;
+
+        /// <summary>
+        /// Event, which is fired when new workspace is added.
```
Comment (ramramps) [inline src/DynamoCore/Interfaces/IDynamoModel.cs]: This event is fired when a new workspace is added.

File: src/DynamoCore/Interfaces/IDynamoModel.cs [hunk 38]
```diff
@@ -10,17 +10,52 @@ namespace Dynamo.Interfaces
...
+        /// </summary>
         event Action<WorkspaceModel> WorkspaceAdded;
+
+        /// <summary>
+        /// Event, which is fired when workspace is removed.
```
Comment (ramramps) [inline src/DynamoCore/Interfaces/IDynamoModel.cs]: This event is fired when a workspace is removed.

File: src/DynamoCore/Interfaces/IDynamoModel.cs [hunk 39]
```diff
@@ -10,17 +10,52 @@ namespace Dynamo.Interfaces
...
+        /// </summary>
         event Action<WorkspaceModel> WorkspaceRemoved;
+
+        /// <summary>
+        /// Event that is fired during the opening of the workspace.
```
Comment (ramramps) [inline src/DynamoCore/Interfaces/IDynamoModel.cs]: This event is fired when a workspace is opened.

File: src/DynamoCore/Interfaces/IDynamoModel.cs [hunk 40]
```diff
@@ -10,17 +10,52 @@ namespace Dynamo.Interfaces
...
+        /// 
+        /// Use the XmlDocument object provided to conduct additional
+        /// workspace opening operations.
```
Comment (ramramps) [inline src/DynamoCore/Interfaces/IDynamoModel.cs]: Remove all this.

File: src/DynamoCore/Logging/DynamoLogger.cs [hunk 41]
```diff
@@ -7,11 +7,25 @@
 
 namespace Dynamo.Logging
 {
+    /// <summary>
+    /// The log level at which to log the message.
```
Comment (ramramps) [inline src/DynamoCore/Logging/DynamoLogger.cs]: Enumerates LogLevel.

File: src/DynamoCore/Logging/DynamoLogger.cs [hunk 42]
```diff
@@ -7,11 +7,25 @@
...
+    /// </summary>
     public enum LogLevel{Console, File, Warning}
+
+    /// <summary>
+    /// The warning level for messages.
```
Comment (ramramps) [inline src/DynamoCore/Logging/DynamoLogger.cs]: Enumerates warning level for log messages.

File: src/DynamoCore/Logging/DynamoLogger.cs [hunk 43]
```diff
@@ -7,11 +7,25 @@
...
+    /// </summary>
     public enum WarningLevel{Mild, Moderate, Error}
 
+    /// <summary>
+    /// Delegate, that takes log event arguments, logs them and return nothing.
```
Comment (ramramps) [inline src/DynamoCore/Logging/DynamoLogger.cs]: This is a delegate used by Log events.

File: src/DynamoCore/Logging/DynamoLogger.cs [hunk 44]
```diff
@@ -7,11 +7,25 @@
...
+    /// </summary>
+    /// <param name="args">Log event arguments.</param>
     public delegate void LogEventHandler(LogEventArgs args);
 
+    /// <summary>
+    /// Event arguments, that are sent to log event handler.
```
Comment (ramramps) [inline src/DynamoCore/Logging/DynamoLogger.cs]: Represents Event arguments that are passed to log event handler.

File: src/DynamoCore/Logging/ILogger.cs [hunk 45]
```diff
@@ -7,10 +7,35 @@ namespace Dynamo.Logging
     /// </summary>
     public interface ILogger
     {
+        /// <summary>
+        /// Logs the specified message.
+        /// </summary>
         void Log(string message);
+
+        /// <summary>
+        /// Logs some data with an associated tag.
```
Comment (ramramps) [inline src/DynamoCore/Logging/ILogger.cs]: remove - some.

File: src/DynamoCore/Migration/Migration.cs [hunk 46]
```diff
@@ -985,8 +991,19 @@ public PortId(string owningNode, int portIndex, PortType type)
             PortType = type;
         }
 
+        /// <summary>
+        /// Node GUID.
+        /// </summary>
         public string OwningNode { get; private set; }
+
+        /// <summary>
+        /// Index
```
Comment (ramramps) [inline src/DynamoCore/Migration/Migration.cs]: Index of a port.

File: src/DynamoCore/Migration/Migration.cs [hunk 47]
```diff
@@ -985,8 +991,19 @@ public PortId(string owningNode, int portIndex, PortType type)
...
+        /// </summary>
         public int PortIndex { get; private set; }
+
+        /// <summary>
+        /// PortType
```
Comment (ramramps) [inline src/DynamoCore/Migration/Migration.cs]: Give example.

File: src/DynamoCore/Models/DynamoModelDelegates.cs [hunk 48]
```diff
@@ -11,6 +11,14 @@ namespace Dynamo.Models
     public delegate void NodeHandler(NodeModel node);
     public delegate void WorkspaceHandler(WorkspaceModel model);   
     public delegate void ActionHandler(Action action);
+
+    /// <summary>
+    /// Delegate, that is mostly used in workspace events.
+    /// When user tries to do something with node.
```
Comment (ramramps) [inline src/DynamoCore/Models/DynamoModelDelegates.cs]: remove : // When user tries to do something with node./// E.g. select it, drag it etc.

File: src/DynamoCore/Models/DynamoModelDelegates.cs [hunk 49]
```diff
@@ -11,6 +11,14 @@ namespace Dynamo.Models
     public delegate void NodeHandler(NodeModel node);
     public delegate void WorkspaceHandler(WorkspaceModel model);   
     public delegate void ActionHandler(Action action);
+
+    /// <summary>
+    /// Delegate, that is mostly used in workspace events.
```
Comment (ramramps) [inline src/DynamoCore/Models/DynamoModelDelegates.cs]: This delegate is used in workspace events.

File: src/DynamoCore/Visualization/DefaultRenderPackageFactory.cs [hunk 50]
```diff
@@ -5,8 +5,16 @@
...
 namespace Dynamo.Visualization
 {
+    /// <summary>
+    /// Example implementation of IRenderPackageFactory.
+    /// DefaultRenderPackageFactory produces DefaultRenderPackages.
+    /// </summary>
     public class DefaultRenderPackageFactory : IRenderPackageFactory
     {
+        /// <summary>
+        /// Tessellation parameters.
```
Comment (ramramps) [inline src/DynamoCore/Visualization/DefaultRenderPackageFactory.cs]: Sets or Returns Tessellation parameters.

File: src/DynamoCore/Visualization/DefaultRenderPackageFactory.cs [hunk 51]
```diff
@@ -17,31 +25,42 @@ public DefaultRenderPackageFactory()
             };
         }
 
+        /// <summary>
+        /// Creates DefaultRenderPackage, the type manufactured by this factory.
+        /// </summary>
```
Comment (ramramps) [inline src/DynamoCore/Visualization/DefaultRenderPackageFactory.cs]: Remove - the type manufactured by this factory

File: src/DynamoCore/Visualization/DefaultRenderPackageFactory.cs [hunk 52]
```diff
@@ -17,31 +25,42 @@ public DefaultRenderPackageFactory()
...
+        /// <returns>DefaultRenderPackage</returns>
         public IRenderPackage CreateRenderPackage()
         {
             return new DefaultRenderPackage();
         }
     }
 
+    /// <summary>
+    /// Example implementation of IRenderPackage.
+    /// DefaultRenderPackage can be used as package for your own visualization.
```
Comment (ramramps) [inline src/DynamoCore/Visualization/DefaultRenderPackageFactory.cs]: remove - own.

File: src/DynamoCore/Visualization/DefaultRenderPackageFactory.cs [hunk 53]
```diff
@@ -108,30 +151,48 @@ public void AddLineStripVertexColor(byte red, byte green, byte blue, byte alpha)
...
+        /// </summary>
+        /// <param name="colors">A buffer of R,G,B,A values corresponding to each vertex.</param>
         public void ApplyMeshVertexColors(byte[] colors)
         {
             meshColors.Clear();
             meshColors.AddRange(colors);
         }
 
+        /// <summary>
+        /// Set a an array of bytes to be used as a color map.
```
Comment (ramramps) [inline src/DynamoCore/Visualization/DefaultRenderPackageFactory.cs]: sets an array of bytes that is used as a color map.

File: src/DynamoCore/Visualization/DefaultRenderPackageFactory.cs [hunk 54]
```diff
@@ -156,96 +217,166 @@ public void Clear()
             DisplayLabels = false;
         }
 
+        /// <summary>
+        /// A tag used to store information about the render package.
+        /// </summary>
         public string Description { get; set; }
 
+        /// <summary>
+        /// A flag indicating whether the render package is selected.
```
Comment (ramramps) [inline src/DynamoCore/Visualization/DefaultRenderPackageFactory.cs]: Returns true if the render package is selected.

File: src/DynamoCore/Visualization/DefaultRenderPackageFactory.cs [hunk 55]
```diff
@@ -156,96 +217,166 @@ public void Clear()
...
+        /// </summary>
         public bool IsSelected { get; set; }
 
+        /// <summary>
+        /// A flag indicating whether the render package has data.
+        /// </summary>
```
Comment (ramramps) [inline src/DynamoCore/Visualization/DefaultRenderPackageFactory.cs]: Returns true if the render package has data.

File: src/DynamoCore/Visualization/DefaultRenderPackageFactory.cs [hunk 56]
```diff
@@ -156,96 +217,166 @@ public void Clear()
...
         public bool HasRenderingData
         {
             get { return pointVertices.Any() || meshVertices.Any() || lineVertices.Any(); }
         }
 
+        /// <summary>
+        /// A flag indicating whether the render package is displaying labels
```
Comment (ramramps) [inline src/DynamoCore/Visualization/DefaultRenderPackageFactory.cs]: Returns true if the render package is displaying labels.

File: src/DynamoCore/Visualization/DefaultRenderPackageFactory.cs [hunk 57]
```diff
@@ -156,96 +217,166 @@ public void Clear()
...
+        /// </summary>
         public bool DisplayLabels { get; set; }
 
+        /// <summary>
+        /// A flag indicating whether the render package requires 
+        /// per vertex coloration.
```
Comment (ramramps) [inline src/DynamoCore/Visualization/DefaultRenderPackageFactory.cs]: Returns true if the render package requires per vertex coloration

File: src/DynamoCore/Visualization/DefaultRenderPackageFactory.cs [hunk 58]
```diff
@@ -156,96 +217,166 @@ public void Clear()
...
+        /// </summary>
         public bool RequiresPerVertexColoration { get; set; }
 
+        /// <summary>
+        /// The number of point vertices in the package.
```
Comment (ramramps) [inline src/DynamoCore/Visualization/DefaultRenderPackageFactory.cs]: Add Returns here. It should be Returns the number of....
correct all the below ones.

File: src/DynamoCore/Visualization/IRenderPackageSource.cs [hunk 59]
```diff
@@ -4,6 +4,10 @@
 
 namespace Dynamo.Visualization
 {
+    /// <summary>
+    /// Interface which provides events, which should be fired when render packages are changed.
```
Comment (ramramps) [inline src/DynamoCore/Visualization/IRenderPackageSource.cs]: This interface has events that is fired when the render packages are changed.

File: src/DynamoCore/Visualization/IRenderPackageFactory.cs [hunk 60]
```diff
@@ -10,6 +10,9 @@ namespace Dynamo.Visualization
     /// </summary>
     public interface IRenderPackageFactory
     {
+        /// <summary>
+        /// Parameters for tessellation.
```
Comment (ramramps) [inline src/DynamoCore/Visualization/IRenderPackageFactory.cs]: Gets or sets the Tessellation parameters.

**Comment by ramramps** [comment]
@aosyatnik - Address the comments. Also. make sure you add proper Returns or Sets every where.

**Comment by aosyatnik** [comment]
Hi, @ramramps , I addressed almost all comments. But I have some questions. I wrote them in file with Priority 1 members. PTAL: https://docs.google.com/spreadsheets/d/1PT8Ha_7764ZP9CH6JKG0Y2do38xT-pDjf1XyEbsqxKs/edit#gid=0

**Comment by ramramps** [comment]
@aosyatnik - OK to remove the "Never used" ones. See the document. 
I would suggest you to run the self service CI before merging.

**Comment by aosyatnik** [comment]
Hi, @ramramps , I just finished running self CI. No fallen tests :)

File: src/DynamoCore/Graph/ModelBase.cs [hunk 61]
```diff
@@ -267,6 +313,11 @@ public XmlElement Serialize(XmlDocument xmlDocument, SaveContext context)
             return element;
         }
 
+        /// <summary>
+        /// Deserialize model from xml node.
+        /// </summary>
+        /// <param name="element">Xml node</param>
+        /// <param name="context"></param>
```
Comment (QilongTang) [inline src/DynamoCore/Graph/ModelBase.cs]: Missing description here?

Comment (ramramps) [inline src/DynamoCore/Graph/ModelBase.cs]: where?

Comment (QilongTang) [inline src/DynamoCore/Graph/ModelBase.cs]: for [context]

**Comment by QilongTang** [comment]
Overall, LGTM.

**Pull request merged**